solr/core/src/java/org/apache/solr/request/SolrRequestInfo.java

```diff
@@ -75,6 +77,9 @@ public static void setRequestInfo(SolrRequestInfo info) {
     } else if (stack.size() > MAX_STACK_SIZE) {
       assert false : "SolrRequestInfo Stack is full";
       log.error("SolrRequestInfo Stack is full");
+    } else if (!stack.isEmpty() && info.req != null) {
+      // if req is null limits will be an empty instance with no limits anyway.
```

Comment:
Maybe add a note that all SRI-s on the stack inherit the first instance of the limits.

Reply:
agreed